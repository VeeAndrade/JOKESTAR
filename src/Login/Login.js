import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { updateUser } from '../Actions'
import './Login.scss';
// import { mapStateToProps, mapDispatchToProps } from '../LandingPage/LandingPage';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      password: ''
    }
  }

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value})
  }

  verifyPassword = e => {
    this.props.updateUser(this.state.username)
    let values = Object.values(this.state);
    values.includes('') ? this.setState({error: true}) : this.clearLogin()
  }

  clearLogin = () => {
    this.setState({username: '', password: '', error: false});
  }

  render() {
    return(
      <section className='login-section'>
        <form className='login-form'>
          <section className='inputs-section'>
            <input 
            type='text'
            placeholder='Username'
            className='username-input'
            name='username'
            value={this.state.username}
            onChange={this.handleChange}
            />
            <input
            type='password'
            placeholder='Password'
            className='password-input'
            name='password'
            value={this.state.password}
            onChange={this.handleChange}
            required
            />
          </section>
          <section className='button-container'>
            {this.state.error !== true ? 
              <Link to='/main-page' className='sign-in-button' onClick={(event) => this.verifyPassword(event)}>
                Sign In
              </Link>
               : console.log(this.state.error)}
          </section>
        </form>
      </section>
    )
  }
}

const mapStateToProps = state => ({
  username: state.username
})

const mapDispatchToProps = dispatch => ({
  updateUser: username => dispatch(updateUser(username))
})

export default connect(mapStateToProps, mapDispatchToProps)(Login);