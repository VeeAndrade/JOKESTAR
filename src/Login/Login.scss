.login-section {
  width: 40%;
  height: 40%;
  border: solid 2px rgb(0, 255, 136);
  background: linear-gradient(rgba(0, 255, 136, 0.384), rgba(0, 255, 136, 0.103));
  margin-top: 6%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.login-form {
  height: 15rem;
  width: 70%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.inputs-section {
  display: flex;
  height: 60%;
  flex-direction: column;
  justify-content: space-around;
}

.password-input,
.username-input {
  height: 2.3rem;
  font-size: 1.1em;
  padding-left: 8px;
  border-radius: 3px;
  border: none;
}

.button-container {
  display: flex;
  height: 40%;
  justify-content: center;
  align-items: center;
}

.sign-in-button {
  height: 2.3rem;
  width: 70%;
  font-size: 1.1em;
  border-radius: 3px;
  border: none;

  &:hover {
    background-color: rgb(0, 255, 136);
  }
}