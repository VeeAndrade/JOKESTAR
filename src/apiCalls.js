export const getRandomJoke = () => {
  return fetch('https://official-joke-api.appspot.com/random_joke')
    .then(response => {
      if (!response.ok) {
        throw Error('200 status code not found: getRandomJoke throw error')
      }
      return response.json()
    }
  )
}

//fetch apis will be defined by the type they are, 
//either single, twopart, or general
//single and two part are the NSFW
//general, programming, 'knock-knock' is safe API