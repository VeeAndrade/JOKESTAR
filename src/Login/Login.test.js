import React from 'react';
import { Login } from './Login';
import { shallow } from 'enzyme';
import { updateUser } from '../apiCalls'

jest.mock('../apiCalls')

describe('Login', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Login
    username={'EvilKitty'}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state('username')).toEqual('')
    expect(wrapper.state('password')).toEqual('')
  });

  it('should change state when handleChange is invoked', () => {
    let mockEvent = {
      target: {
        name: 'username',
        value: 'EvilKitty'
      }
    }
    let mockEvent2 = {
      target: {
        name: 'password',
        value: '123456'
      }
    }
    expect(wrapper.state('username')).toEqual('')
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('username')).toEqual('EvilKitty')

    expect(wrapper.state('password')).toEqual('')
    wrapper.instance().handleChange(mockEvent2);
    expect(wrapper.state('password')).toEqual('123456')
  });

  // it('should verify the password when verifyPassword is invoked', () => {
  //   let mockEvent2 = {
  //     target: {
  //       name: 'password',
  //       value: '123456'
  //     }
  //   }
  //   updateUser.mockImplementation(() => {
  //     return Promise.resolve([{ username: 'KillerKitty'}])
  //   })
  //   wrapper.instance().verifyPassword(mockEvent2);
  //   expect(wrapper.state('error')).toEqual(true)
  // })
}); 