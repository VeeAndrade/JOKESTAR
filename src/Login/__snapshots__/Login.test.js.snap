// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<section
  className="login-section"
>
  <form
    className="login-form"
  >
    <section
      className="inputs-section"
    >
      <input
        className="username-input"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        className="password-input"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </section>
    <section
      className="button-container"
    >
      <Link
        className="sign-in-button"
        onClick={[Function]}
        to="/main-page"
      >
        Sign In
      </Link>
    </section>
  </form>
</section>
`;
